component/helper/src/main/java/org/wso2/carbon/extension/identity/helper/util/IdentityHelperUtil.java

```diff
@@ -344,4 +351,34 @@ public static void loadApplicationAuthenticationXMLFromRegistry(AuthenticationCo
                 .getAuthenticatorBean(authenticatorName);
         return authConfig.getParameterMap();
     }
-}
\ No newline at end of file
+
+    /**
+     * Append a query param map to the URL (URL may already contain query params).
+     *
+     * @param url         URL string to append the params.
+     * @param queryParams Map of query params to be append.
+     * @return Built URL with query params.
+     * @throws UnsupportedEncodingException Throws when trying to encode the query params.
+     */
+    public static String appendQueryParamsToUrl(String url, Map<String, String> queryParams)
+            throws UnsupportedEncodingException {
+
+        if (StringUtils.isEmpty(url)) {
+            throw new IllegalArgumentException("Passed URL is empty.");
+        }
+        if (queryParams == null) {
+            throw new IllegalArgumentException("Passed query param map is empty.");
+        }
+
+        List<String> queryParam1 = new ArrayList<>();
```

Comment:
Shall we change the name to encodedQueryParamList

Reply:
+1. Updated the name